### PR TITLE
`gen_stubs`: Add complete coverage for all (non-ILP32) arches.

### PR DESCRIPTION
**Pull request by alexrp**
In particular, we were missing arm, m68k, and s390x.

### CODE REVIEW

File: tools/gen_stubs.zig [hunk 1]
```diff
@@ -593,16 +624,19 @@ fn parseElf(parse: Parse, comptime is_64: bool, comptime endian: builtin.Endian)
 fn archIndex(arch: std.Target.Cpu.Arch) u8 {
     return switch (arch) {
         // zig fmt: off
-        .riscv64     => 0,
```
Comment (alexrp) [inline tools/gen_stubs.zig:596]: I'm unsure if the the indexes that were assigned here previously had significance. If they did, I can revert this part and add the new arches at 10-12.

**Comment by andrewrk** [comment]
Those are missing because I was unable to get them to build: https://github.com/ziglang/zig/wiki/Updating-libc#updating-the-libcs-file

**Comment by alexrp** [comment]
Ok, I'll try those and see if they work now. At least the s390x error should be fixed by #20451. Not sure about arm and m68k.

**Comment by alexrp** [comment]
* arm still fails with the error you got.
* m68k doesn't segfault anymore, but: `./configure: error: unsupported long double type`
* s390x compiles (after removing enough `f16` stuff from compiler-rt), but fails to link due to: `ld.lld: error: relocation R_390_PC32DBL cannot be used against symbol '__tls_get_addr'; recompile with -fPIC`

Well, I guess let's leave this PR until the LLVM 19 upgrade and then re-evaluate?

**Comment by alexrp** [comment]
Closing for now; will retry the above after we upgrade to LLVM 19, and then reopen if needed.

**Pull request closed without merging**